### PR TITLE
[NOD-656] Log hashrate in kaspaminer

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: cmd/kaspaminer/mineloop.go [hunk 1]
```diff
@@ -50,6 +52,19 @@ func mineLoop(client *minerClient, numberOfBlocks uint64, blockDelay uint64) err
 		doneChan <- struct{}{}
 	})
 
+	spawn(func() {
+		lastCheck := time.Now()
+		for range time.Tick(time.Second) {
```
Comment (svarogg) [inline cmd/kaspaminer/mineloop.go]: Make it 10 seconds.

File: cmd/kaspaminer/mineloop.go [hunk 2]
```diff
@@ -50,6 +52,19 @@ func mineLoop(client *minerClient, numberOfBlocks uint64, blockDelay uint64) err
...
+			currentHashesTried := hashesTried
+			currentTime := time.Now()
+			hashRate := float64(currentHashesTried) / currentTime.Sub(lastCheck).Seconds()
+			log.Infof("Current hash rate is %.2f hashes per second", hashRate)
```
Comment (svarogg) [inline cmd/kaspaminer/mineloop.go]: Use KHashes instead

File: cmd/kaspaminer/mineloop.go [hunk 3]
```diff
@@ -50,6 +52,20 @@ func mineLoop(client *minerClient, numberOfBlocks uint64, blockDelay uint64) err
 		doneChan <- struct{}{}
 	})
 
+	spawn(func() {
```
Comment (stasatdaglabs) [inline cmd/kaspaminer/mineloop.go]: Please move all of this to a separate function.

File: cmd/kaspaminer/mineloop.go [hunk 4]
```diff
@@ -50,6 +52,20 @@ func mineLoop(client *minerClient, numberOfBlocks uint64, blockDelay uint64) err
...
+		lastCheck := time.Now()
+		for range time.Tick(10 * time.Second) {
```
Comment (stasatdaglabs) [inline cmd/kaspaminer/mineloop.go]: Please move 10 * time.Second to a constant.

**Pull request merged**